Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/securitycontextconstraints/provider_test.go

```diff
@@ -0,0 +1,266 @@
+		"failCapsSCC": {
+			pod:           failCapsPod,
+			scc:           defaultSCC(),
+			expectedError: "Capability is not allowed to be added",
+		},
+		"failHostDirSCC": {
+			pod:           failHostDirPod,
+			scc:           defaultSCC(),
+			expectedError: "Host Volumes are not allowed to be used",
+		},
+	}
+
+	for k, v := range errorCases {
+		provider, err := NewSimpleProvider(v.scc)
+		if err != nil {
+			t.Fatal("unable to create provider %v", err)
+		}
+		errs := provider.ValidateSecurityContext(v.pod, &v.pod.Spec.Containers[0])
+		if len(errs) == 0 {
+			t.Errorf("%s expected validation failure but did not receive errors", k)
+			continue
+		}
+		if !strings.Contains(errs[0].Error(), v.expectedError) {
+			t.Errorf("%s received unexpected error %v", k, errs)
+		}
+	}
+}
+
+func TestValidateSuccess(t *testing.T) {
+	defaultSCC := func() *api.SecurityContextConstraints {
+		return &api.SecurityContextConstraints{
+			ObjectMeta: api.ObjectMeta{
+				Name: "scc-sa",
+			},
+			RunAsUser: api.RunAsUserStrategyOptions{
+				Type: api.RunAsUserStrategyRunAsAny,
+			},
+			SELinuxContext: api.SELinuxContextStrategyOptions{
+				Type: api.SELinuxStrategyRunAsAny,
+			},
+		}
+	}
+
+	var notPriv bool = false
+	defaultPod := func() *api.Pod {
+		return &api.Pod{
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						SecurityContext: &api.SecurityContext{
+							// expected to be set by defaulting mechanisms
+							Privileged: &notPriv,
+							// fill in the rest for test cases
+						},
+					},
+				},
+			},
+		}
+	}
+
+	// fail user strat
+	userSCC := defaultSCC()
+	var uid int64 = 999
+	userSCC.RunAsUser = api.RunAsUserStrategyOptions{
+		Type: api.RunAsUserStrategyMustRunAs,
+		UID:  &uid,
+	}
+	userPod := defaultPod()
+	userPod.Spec.Containers[0].SecurityContext.RunAsUser = &uid
+
+	// fail selinux strat
+	seLinuxSCC := defaultSCC()
+	seLinuxSCC.SELinuxContext = api.SELinuxContextStrategyOptions{
+		Type: api.SELinuxStrategyMustRunAs,
+		SELinuxOptions: &api.SELinuxOptions{
+			Level: "foo",
+		},
+	}
+	seLinuxPod := defaultPod()
+	seLinuxPod.Spec.Containers[0].SecurityContext.SELinuxOptions = &api.SELinuxOptions{
+		Level: "foo",
+	}
+
+	privSCC := defaultSCC()
+	privSCC.AllowPrivilegedContainer = true
+	privPod := defaultPod()
+	var priv bool = true
+	privPod.Spec.Containers[0].SecurityContext.Privileged = &priv
+
+	capsSCC := defaultSCC()
+	capsSCC.AllowedCapabilities = []api.Capability{"foo"}
+	capsPod := defaultPod()
+	capsPod.Spec.Containers[0].SecurityContext.Capabilities = &api.Capabilities{
+		Add: []api.Capability{"foo"},
+	}
+
+	hostDirSCC := defaultSCC()
+	hostDirSCC.AllowHostDirVolumePlugin = true
+	hostDirPod := defaultPod()
+	hostDirPod.Spec.Volumes = []api.Volume{
+		{
+			Name: "bad volume",
+			VolumeSource: api.VolumeSource{
+				HostPath: &api.HostPathVolumeSource{},
+			},
+		},
+	}
+
+	errorCases := map[string]struct {
+		pod *api.Pod
+		scc *api.SecurityContextConstraints
+	}{
+		"userSCC": {
```

Comment:
You might consider giving these cases more descriptive names -- it's okay for them to be a little long.